lib/awful/layout/suit/tile.lua.in

```diff
@@ -156,26 +155,28 @@ local function tile_group(cls, wa, orientation, fact, group)
     size = math.min(size, available)
 
     local coord = wa[y]
-    local geom = {}
     local used_size = 0
     local unused = wa[height]
     for c = group.first,group.last do
         local i = c - group.first +1
-        geom[width] = size - cls[c].border_width * 2
-        geom[height] = math.floor(unused * fact[i] / total_fact) - cls[c].border_width * 2
+        local geom = {}
+        geom[width] = size
+        geom[height] = math.floor(unused * fact[i] / total_fact)
         geom[x] = group.coord
         geom[y] = coord
-        geom = cls[c]:geometry(geom)
```

Comment:
Actually I just found a C function that does exactly what I want: [`client_apply_size_hints`](https://github.com/awesomeWM/awesome/blob/master/objects/client.c#L605-708). You give it geometry and it changes the geometry based on the size hints without actually updating the client. Perhaps this should be exposed to Lua?